### PR TITLE
fix(model): added meaningful fields for NN builds, other patches

### LINKED ISSUE #1: fix(model): adding meaningful fields for neural network builds
# Description

Currently, the synthetic data used for model training, with around `720` Records only brings out `0.94` Accuracy Score, and it is in XGBoost, and with tweaks, is in Linear Regression. While the score seems above average, the results for the purchase prediction were totally off, partly due the amount of noise with the data. 

- Populate the synthetic data with meaningful fields, such as
    - `base_price`: Will be used by the model to compute price ratio (Discounts/ Premium Prices)
    - `time_of_day_bucket`: Corresponds to `Overnight`/`Morning`/`Afternoon`/`Evening`
    - `popularity_factor`: Internally computed by the device popularity, category popularity, and mostly `rating`.

### PR DESCRIPTION
**Pull request by niyasrad**
# Description

- Populated the synthetic data with meaningful fields, such as
    - `base_price`: Will be used by the model to compute price ratio (Discounts/ Premium Prices)
    - `time_of_day_bucket`: Corresponds to `Overnight`/`Morning`/`Afternoon`/`Evening`
    - `popularity_factor`: Internally computed by the device popularity, category popularity, and mostly `rating`
- Patched the API URL for retrieving sales data from `sales/sales` to `sales/`
- Patched `MindsDB` values in the helm deployment deployment to hold more CPU, Memory to prevent OOM .

## Summary by CodeRabbit

- **New Features**
  - Streamlined the sales submission endpoint for simpler access.
  - Enhanced sales analytics with refined metric calculations and expanded data attributes, including improved rating ranges, pricing logic, and time-of-day categorization.

- **Chores**
  - Increased resource allocations for the analytics component to boost overall performance.

### CODE REVIEW

File: backend/app/utils/sales/synthesise.py [hunk 1]
```diff
@@ -30,22 +30,25 @@ def generate_timestamp(hours_ago: int) -> str:
...
-        rating (float): The rating of the product, ranging from 3.0 to 5.0.
+        rating (float): The rating of the product, ranging from 2.0 to 5.0.
 
     Returns:
         float: The computed popularity factor.
     """
-    device_popularity = random.uniform(0.9, 1.1)  # reduced range for stability
-    return category_popularity * (rating / 5) * device_popularity
+    device_popularity = random.uniform(0.95, 1.05) * category_popularity
+    rating_factor = ((rating - 2) / 3) ** 0.9 * 1.8
```
[the remark this answers is not in the document]
Comment (niyasrad) [reply backend/app/utils/sales/synthesise.py:50]: I believe this was made to ensure that the rating factor stands out, with a bit of noise. Resolving this.
